### ISSUE TITLE
Blogs I Follow Widget

### ISSUE THREAD

**Issue by mrjarbenne** [r-a-y/buddypress-followers#64]
Enhancement request: A widget (similar to the existing Users I Follow widget) that would allow me -- on my subdomain in a multisite instance -- to display the blogs I follow on the network. This widget should allow me to identify how many blogs to list in the widget.

Cheers.

**Comment by espellcaste** [comment]
+1

**Comment by r-a-y** [comment]
The one issue is blogs do not have avatars yet.  Do we want to list blogs with the user avatars instead?

Something like:

> Blogs I Follow
> * [USER AVATAR] TITLE OF BLOG
> * [USER AVATAR] TITLE OF BLOG
> * [USER AVATAR] TITLE OF BLOG

Instead of a traditional avatar-only widget?

Blog avatars might make BP 2.3.  Even if blog avatars makes 2.3, would we want to use an avatar-only widget?  If so, perhaps use a default blog avatar icon?

Feedback appreciated.

**Comment by espellcaste** [comment]
I can see a lot of scenarios here:

Client wants avatar for blogs;
Client wants avatar of the user, not of the blog;
Client want default avatar, as a default (maybe he's gonna use a custom avatar of his).

If I were to choose, I'd choose the default one or the avatar for Blogs. 

What do you think about allowing or disabling one or the other with a function?

**Comment by mrjarbenne** [comment]
I think you need the title of the blog regardless of the avatar conversation. 

With the Avatar, a default icon makes sense for me, until such time as there is a blog avatar option. Even at that point having a default avatar specific to blogs, in cases where the user has not uploaded a blog avatar, makes good sense. 

It could be something as simple as a dashicon, so the entire widget becomes more text based rather than avatar based.

https://developer.wordpress.org/resource/dashicons/#welcome-write-blog

**Comment by mrjarbenne** [comment]
On another note around design, I think it would also be helpful to be able to customize the widget title. Blogs I Follow may not apply to the context a user might be using the widget for. Being able to change that to "Sites I Read" "inspirational sites" "My Classes Blogs" etc would be nice.

**Comment by r-a-y** [comment]
Yeah, I plan on making the widget title a configurable option.  That, and the user ID (or perhaps username?) to fetch the followed sites for.

**Comment by mrjarbenne** [comment]
Now that site icons is a core feature (WP 4.3) I think we now have our avatar source.
